src/backend/marsha/core/api/video.py

```diff
@@ -686,3 +689,77 @@ def end_sharing(
         serializer = self.get_serializer(video)
         channel_layers_utils.dispatch_video_to_groups(video)
         return Response(serializer.data)
+
+    @action(
+        methods=["patch"],
+        detail=True,
+        url_path="start-recording",
+        permission_classes=[
+            permissions.IsTokenResourceRouteObject
+            & (permissions.IsTokenInstructor | permissions.IsTokenAdmin)
+        ],
+    )
+    # pylint: disable=unused-argument
+    def start_recording(
+        self,
+        request,
+        pk=None,
+    ):
+        """Starts video recording.
+
+        Parameters
+        ----------
+        request : Type[django.http.request.HttpRequest]
+            The request on the API endpoint
+        pk: string
+            The primary key of the video
+
+        Returns
+        -------
+        Type[rest_framework.response.Response]
+            HttpResponse with the serialized video.
+        """
+        video = self.get_object()
+        try:
+            video.start_recording()
+        except VideoRecordingError as error:
+            return Response({"detail": str(error)}, status=400)
+        serializer = self.get_serializer(video)
+        return Response(serializer.data)
+
+    @action(
+        methods=["patch"],
+        detail=True,
+        url_path="end-recording",
+        permission_classes=[
+            permissions.IsTokenResourceRouteObject
+            & (permissions.IsTokenInstructor | permissions.IsTokenAdmin)
+        ],
+    )
+    # pylint: disable=unused-argument
+    def end_recording(
```

Comment:
As `remote_control` is not an action, `self.get_object()` and `self.get_serializer()` won't work.
Passing a video as a parameter would be needed.
Also, as it's not an action, I don't think it should return a `Response` object.
So `remote_control` would just be a wrapper around `video.start_recording` and `video.end_recording`, leaving `get_object`, `get_serializer` and `except VideoRecordingError` in each enpoints.
I don't think this wrapper would offer any advantage, and it would add useless complexity.